src/main/scala/com/crobox/clickhouse/stream/ClickhouseIndexingSubscriber.scala

```diff
@@ -279,35 +295,70 @@ class ClickhouseBulkActor(targetTable: String,
     buffer.clear()
     context.stop(self)
   }
-
-  val insertQuery = s"INSERT INTO $targetTable FORMAT JSONEachRow"
-
   private val objectMapper = new ObjectMapper()
     .registerModule(new JodaModule)
     .registerModule(new DefaultScalaModule)
     .setSerializationInclusion(JsonInclude.Include.NON_DEFAULT)
 
   private def index(): Unit = {
-    val payload = buffer.map(i => objectMapper.writeValueAsString(i.data)).mkString("\n") + "\n"
-    val count   = buffer.size
+    val colList = buffer
```

Comment:
This is a simple but IMO fine implementation that just takes the cols from the biggest element (to cope with empty entries). Should be working as long as all elements have the same shape. If not, then clickhouse will respond with an exception, and it does a good job at it.